Makefile

```diff
@@ -507,24 +502,21 @@ KBUILD_CFLAGS   := -Wall -Wundef -Werror=strict-prototypes -Wno-trigraphs \
 		   -Werror=return-type -Wno-format-security \
 		   -std=gnu89
 KBUILD_CPPFLAGS := -D__KERNEL__
-KBUILD_RUSTCFLAGS :=
-# TODO: a simple way to update `Cargo.lock` when we add a new driver
-KBUILD_CARGOFLAGS := $(CARGO_VERBOSE) --locked \
-		    -Z build-std=core,alloc -Z unstable-options \
-		    --target=$(srctree)/arch/$(SRCARCH)/rust/target.json
+KBUILD_RUSTCFLAGS := --emit=dep-info,obj,link -Z binary_dep_depinfo=y \
```

Comment:
Why are both `obj` and `link` used? Only one or the other should be necessary.

Reply:
`obj` is used to get the object files, `link` to get the `.rlib`s so that Rust code can depend on Rust code. We can indeed get the object files from the `.rlib`s, but it seemed easier this way.

Reply:
Would `--emit=dep-info,obj,metadata` work? Then you can pass the `.rmeta` files to rustc and the `.o` files to the linker. This would save disk space and make compilation a bit faster due to no longer having to create an ar archive with the object file and metadata (the `.rlib`).

Reply:
It should work, yeah (the only difference I recall was that the archives could have several object files rather than a merged; but it should not matter). I will give it a try +1

Reply:
It seems `rustc` requires a fix -- filed https://github.com/rust-lang/rust/issues/81117.

Reply:
Should be fixed when https://github.com/rust-lang/rust/issues/81118 goes in.

Reply:
https://github.com/rust-lang/rust/commit/f9275e1092232fcb8ec117fc4acca990f57cba15 is now in -- I will give it a go when we get a nightly. I didn't see any significant space reduction when I tried locally (the metadata is the biggest part anyway, by far), but nevertheless avoiding archives is cleaner for us.

Reply:
#75